ath79: use rgmii-id instead of rgmii in ubnt,lap-120

rogerpueyo opened the pull request:
~~Since commit 4b2ab533d4 (ath79: disable delays on AT803X config init)~~

Since commit 6f2e1b7485 (ath79: disable delays on AT803X config init)
the incoming incoming traffic on the ubnt,lap-120 devices Ethernet
port was not making it through. After it, the working GMAC rxd and
rxdv delay values changed from 3 to 2.

Fixes [FS#2893](https://bugs.openwrt.org/index.php?do=details&task_id=2893).

Signed-off-by: Roger Pueyo Centelles <roger.pueyo@guifi.net>

rogerpueyo committed:
ath79: use rgmii-id instead of rgmii in ubnt,lap-120

Since commit 6f2e1b7485 (ath79: disable delays on AT803X config init)
the incoming incoming traffic on the ubnt,lap-120 devices Ethernet
port was not making it through. Using rgmii-id instead of rgmii (same
configuration as ubnt,litebeam-ac-gen2) fixes it.

Fixes FS#2893.

Signed-off-by: Roger Pueyo Centelles <roger.pueyo@guifi.net>

xback commented:
Hi Roger,

When I click on the commit in your description I end uo at a kernel bump commit?

Thanks,

Koen

rogerpueyo commented:
Commit *6f2e1b7485*, wrong copy-paste!

xback commented:
@blocktrron 
Can you check this one please?

Thanks!

blocktrron commented on target/linux/ath79/dts/ar9342_ubnt_lap-120.dts:
```diff
@@ -31,8 +31,8 @@
-		rxdv-delay = <3>;
+		rxd-delay = <2>;
+		rxdv-delay = <2>;
 	};
```
You are changing the delays introduced by the MAC - the patch changed the delays introduced by the PHY.

In the end it doesn't matter. Have you verified the link works on every link speed without dropping packets?

rogerpueyo replied:
Yes, good point.

Before changing the delay values, none of the speeds worked (10/100/1000). With this change, I see now it only works for 1000 Mbps. Where to go next? pll-data?

I've also seen that setting rxd and rxdv to <1> also works but, again, only for 1000 Mbps.

blocktrron replied:
Did it work previously for all link speeds? If so - you can try out `rgmii-id`, `rgmii-rxid`, `rgmii-txid` on the MACs (not the PHYs!) `phy-mode` property and leave the GMAC delays untouched. One of the should reflect the previous state.

rogerpueyo replied:
Yes, it did work at the three speeds, 10, 100 and 1000 Mbps, with no packet loss.

The stock firmware states RGMII
```
[   36.535000] ATH_MAC_TIMER: enet unit:0 is up (RGMii 10Mbps half duplex)
[   36.535000] ATH_MAC_TIMER: done cfg2 0x7114 ifctl 0x0 miictrl  
[   36.535000] ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
```

I'm trying to find the right combination. `rgmii-id` works for the three speeds, but packet losses appear on 10 and 100 Mbps. `rgmii-rxid` and `rgmii-txid` work for 10 and 100 Mbps, but not for 1 Gbps.

Should I better test the delays or the pll-data?

rogerpueyo replied:
> I'm trying to find the right combination. `rgmii-id` works for the three speeds, but packet losses appear on 10 and 100 Mbps. `rgmii-rxid` and `rgmii-txid` work for 10 and 100 Mbps, but not for 1 Gbps.

Great, rgmii-id works correctly at all speeds; my USB network adapter was causing the retransmissions.

rogerpueyo commented:
Hi @blocktrron ,

OK, so I went through a lot of testing different configurations and, so far, this the one that works the best. I switched to rgmii-id instead of rgmii (like the LiteBeam AC Gen2) and not only it works in all speeds (1000/100/10), but I also got rid of all the retransmissions. It also had to do with the USB network card I was using, which took me a while to realise that it was causing the retransmissions :facepalm:.

Fixed, hopefully.

adrianschmutzler commented:
@blocktrron Can you please reevaluate this PR?

ynezz commented:
Have you tested this on 5.4 as well?

rogerpueyo commented:
Hi,

Not yet. I'll be testing it during this week and report here.

rogerpueyo commented:
> Have you tested this on 5.4 as well?

I've just tested it, it works correctly on 5.4. Current snapshot does not work; the patch is still needed.

Rebased master and force-pushed.

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git
